Switch from monty to orjson for serialization

munrojm opened the pull request:
This PR switches to using orjson for custom serialization in ReadOnlyResource.

shyamd commented:
Why `orjson`? 

It seems silly, but can you add a test for `custom_serialization`. Maybe rename in `object_id_serilaization_helper` to make it clearer what it is doing?

Also, should the flag in `ReadOnlyResource` be called something different like `disable_validation`?

munrojm commented:
`orjson` is extremely fast and can handle `datetime` object natively while allowing for extensions for things like bson types.

I have renamed things as you have suggested and added the test.
